contracts/generator_controller/src/contract.rs

```diff
@@ -136,6 +140,75 @@ pub fn execute(deps: DepsMut, env: Env, info: MessageInfo, msg: ExecuteMsg) -> E
     }
 }
 
+/// ## Description
+/// This function removes all votes applied by blacklisted voters. Returns [`Response`] in case
+/// of success or [`ContractError`] in case of errors.
+///
+/// ## Params
+/// * **deps** is an object of type [`DepsMut`].
+///
+/// * **env** is an object of type [`Env`].
+///
+/// * **holders** is a vector of type [`String`]. Contains blacklisted holders whose votes will be
+/// removed.
+fn kick_blacklisted_voters(deps: DepsMut, env: Env, voters: Vec<String>) -> ExecuteResult {
```

Comment:
imho, there should be some constant `voters` limit to avoid gas limitation error